File: resources/[core]/dg-core/import.lua
if GetCurrentResourceName() == 'dg-core' then
    function GetSharedObject()
        return DGCore
    end

    exports('GetSharedObject', GetSharedObject)
end

if GetResourceState('dg-core') == 'started' then
  DGCore = exports['dg-core']:GetSharedObject()
end

File: resources/[core]/dg-core/import.js
if (GetResourceState('dg-core') == 'started') {
  (async () => {
    while (exports?.['dg-core']?.GetSharedObject === undefined) {
      await new Promise(r => setTimeout(r, 100));
    }
  })();
  global.DGCore = exports['dg-core'].GetSharedObject();
}

on('onResourceStart', async res => {
  if (res == 'dg-core') {
    while (exports?.['dg-core']?.GetSharedObject === undefined) {
      await new Promise(r => setTimeout(r, 100));
    }
    global.DGCore = exports['dg-core'].GetSharedObject();
  }
});
